pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -38,6 +38,12 @@ func NewFlowRecords() *FlowRecords {
 
 // AddOrUpdateFlowRecord adds or updates the flow record in the record map given the connection.
 func (fr *FlowRecords) AddOrUpdateFlowRecord(key flowexporter.ConnectionKey, conn flowexporter.Connection) error {
+	// If the connection is in dying state and corresponding flow records have been exported,
+	// there is no need to add or update the record.
+	if flowexporter.IsConnectionDying(&conn) && conn.IsExported {
```

Comment:
Could we use the `AreRecordsDeleted` flag here? I think it will work unless I am missing something. I do not see any other usage for `IsExported`, so it can be removed.
`AreRecordsDeleted` name could be changed to something simpler.. `DoneExport`? What do you think?

Reply:
Sure. Updated.